tla/loc.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import dataclasses
+import sys
+
+@dataclasses.dataclass
+class LOC:
+    code: int = 0
+    comment: int = 0
+    blank: int = 0
+
+    @property
+    def total(self):
+        return sum(dataclasses.astuple(self))
+
+    def __str__(self):
+        return f"code: {self.code} comments: {self.comment} blank: {self.blank} total: {self.total}"
+
+    def __add__(self, other):
+        return LOC(
+            self.code + other.code,
+            self.comment + other.comment,
+            self.blank + other.blank
+        )
+
+def count_loc(lines) -> LOC:
+    loc = LOC()
+    for line in lines:
+        line_no_ws = line.strip()
+        if not line_no_ws:
+            loc.blank += 1
+        elif line_no_ws.startswith("\*"):
```

Comment:
I know this is quick-and-dirty, but listing some obvious improvements to accuracy:

- Lines starting `(*` are comments
- Lines starting `----` are comments
- Lines starting `====` are comments
- Everything _after_ any `====`-starting line is a comment

(I think TLA might also say anything pre-module is a comment? I'm not sure, but I don't think we do that. We definitely do the latter, eg in `SIMccfraft.tla`)

Reply:
Thank you, will update. Some of the lines starting with `----` are arguably code, the module name declaration at the top for example, is interpreted by TLC. Not sure about pre-module lines, I will check.